CMakeLists.txt

```diff
@@ -79,7 +79,7 @@ macro(clangformat_targets)
     message(WARNING "Not generating clangformat target, no clang-format tool found")
   else ()
     message(STATUS "Generating clangformat target using ${CLANG_FORMAT}")
-    file(GLOB_RECURSE ALL_SOURCE_FILES *.cc *.h *.hh)
+    file(GLOB_RECURSE ALL_SOURCE_FILES src/*.cc src/*.h src/*.hh)
```

Comment:
Not that I object, but why the change?

Reply:
I have about 20 different build directories inside, and this was searching all of them looking for something to format, and the WSL perf was annoying.

Reply:
Putting your build directory outside your source directory avoids a lot of that and also means that you can use a faster FS for the builds (e.g. with WSL2, put the sources in the share FS but the build directories in a Linux native FS).  I typically do builds from a directory on a ZFS FS that has sync disabled (if the machine crashes during a build and I lose some files in my build directory, I don't care).

I'm not sure this glob is actually correct.  Does this find files in `src/{whatever}/*.h`?

Reply:
It is correct:

```
Examples of recursive globbing include:

/dir/*.py  - match all python files in /dir and subdirectories
```
https://cmake.org/cmake/help/latest/command/file.html#filesystem